### PR TITLE
add --amazonec2-keypair-name flag, fix #661

### LINKED ISSUE #661: keypair prevents creation of machine with same name as the one just removed
If the create command fails (in this case because I forgot the vpc-id flag) and I remove the failed image, I can't create a new one with the same name because the keypair doesn't get deleted:

```console
$ docker-machine rm -f demo
$ docker-machine create --driver=amazonec2 --amazonec2-vpc-id=vpc-foo demo
INFO[0000] Creating CA: /Users/dan/.docker/machine/certs/ca.pem 
INFO[0001] Creating client certificate: /Users/dan/.docker/machine/certs/cert.pem 
ERRO[0003] Error creating machine: There is already a keypair with the name demo.  Please either remove that keypair or use a different machine name.
```

It seems that `rm -f` should remove associated keypairs.

Related: #119.

### PR DESCRIPTION
**Pull request by vikstrous**
Signed-off-by: Viktor Stanchev <me@viktorstanchev.com>

### CODE REVIEW

**Comment by nathanleclaire** [comment]
Have you tried master?  https://github.com/docker/machine/pull/3362 was intended to solve this issue

**Comment by vikstrous** [comment]
My problem wasn't that keys don't get deleted sometimes. My bigger problem was that, if they don't, you can't create the machine again. I was also extremely annoyed that our AWS account has hundreds of ssh keys for no reason because of docker machine.

**Comment by sebadiaz** [comment]
+1

**Comment by lucasm-iRonin** [comment]
+1

File: drivers/amazonec2/amazonec2.go [hunk 1]
```diff
@@ -383,19 +397,36 @@ func (d *Driver) DriverName() string {
...
 			return err
 		}
 	}
 
+	// In case we got a result with an empty set of keys
 	if err == nil && len(key.KeyPairs) != 0 {
-		return fmt.Errorf("There is already a keypair with the name %s.  Please either remove that keypair or use a different machine name.", d.MachineName)
+		if !keyShouldExist {
+			return fmt.Errorf("There is already a keypair with the name %s.  Please either remove that keypair or use a different machine name.", d.MachineName)
+		} else {
```
Comment (nathanleclaire) [inline drivers/amazonec2/amazonec2.go]: I think I would prefer to just leave this block out if it has no code content

**Comment by nathanleclaire** [comment]
It seems this problem still persists for users -- @vikstrous if you will fix the one nit I mentioned, and also update the documentation, I'll merge this PR.  Thanks

**Comment by vikstrous** [comment]
updated

**Comment by nathanleclaire** [comment]
test this please

**Comment by vikstrous** [comment]
this should be ready to merge, right?

**Comment by nathanleclaire** [comment]
Yup Docs CI caught up. LGTM and thanks @vikstrous !

**Comment by SvenDowideit** [comment]
@nathanleclaire is this for v0.8.1 or v0.9.0? - if you set the milestone docs'll know when to publish it (and if its a fix for older releases, use that as the milestone?)

**Pull request merged**